src/librustc_typeck/check/mod.rs

```diff
@@ -831,18 +831,22 @@ fn primary_body_of(
 }
 
 fn has_typeck_tables(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
+    // FIXME(#71104) some `LocalDefId` do not seem to have a corresponding `HirId`.
+    if let Some(def_id) = def_id.as_local() {
+        if tcx.hir().opt_local_def_id_to_hir_id(def_id).is_none() {
+            return false;
+        }
+    }
+
```

Comment:
Interesting...

Reply:
Anyway the thing is that this function already had a hack, it just didn't happen early enough, before letting the `DefId` reach any other code. Maybe the hack should be moved to its callers, idk.